### ISSUE TITLE
Support APIGW Models in API Events

### ISSUE THREAD

**Issue by keetonian** [awslabs/serverless-application-model#895]
It would be nice to add options to the `API` event source for `AWS::Serverless::Function` for models (request and response) and validators similar to how authorizers can be defined in there now without having to dive into swagger.

_Originally posted by @beck3905 in https://github.com/awslabs/serverless-application-model/issues/850#issuecomment-485141374_

Models in swagger: https://swagger.io/docs/specification/2-0/describing-request-body/ (see Object Payload section).

I think we should add models to the `Serverless::Api` resource and allow events to reference them:

```yaml
Resource:
  Api:
    Type: AWS::Serverless::Api
    Properties:
      Models: # use swagger format to define models. This gets passed through to the `definitions` section of swagger.
        User:   
          type: object
          required:
            - userName
          properties:
            userName:
              type: string
            firstName:
              type: string
            lastName:
              type: string
     ...

  Function:
    Type: AWS::Serverless::Function
    Properties:
      Events:
        ApiEvent:
          Type: Api
          Properties:
            RestApiId: !Ref Api
            RequestModel: User    # reference by string name
            ResponseModel: User  # reference by string name
```

**Comment by beck3905** [comment]
@keetonian Thanks for creating this feature request.  I may try to take a stab at it if I have time in the next month or so.

**Comment by beck3905** [comment]
@keetonian I think this new feature request I created is related to this existing one: https://github.com/awslabs/serverless-application-model/issues/931.  Essentially, I envision adding optional properties to the API event source definition the correspond to the properties available on the AWS::ApiGateway::Method resource type: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-method.html.  Then those properties if provided would be translated into the swagger definition for that method.  What are your thoughts on this?

**Comment by keetonian** [comment]
I think that this is a great idea and would be a good feature! Thank you for bringing this up.

**Comment by praneetap** [comment]
Released with v1.13.0
